### PR TITLE
add threading tests

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
Adds tests for #1210.

Note, turns out testing coverage of stuff running in another thread is tricky.  So in addition to actually running stuff in the other thread, I run the workers in the main thread here to test various methods.

## Type of change

- [x] tests

# How has this been tested?

- [x] the test suite for my feature covers the qt.threading module
- [x] all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: napari/_qt/_tests/test_threading.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+# this is just for the sake of coverage
+def test_thread_worker_in_main_thread():
+    """Test basic threadworker on a function"""
+
+    def func(x):
+        return x
+
+    thread_func = threading.thread_worker(func)
+    worker = thread_func(2)
+    assert worker.work() == 2
```
Comment (jni) [inline napari/_qt/_tests/test_threading.py]: What exactly in this code is causing it to be run on the main thread?

Comment (tlambert03) [reply napari/_qt/_tests/test_threading.py]: the fact that we call `worker.work()` instead of `worker.start()`... will add that to the docs

**Review by jni — APPROVED** [review]
@tlambert03 I asked a question but mostly for curiosity. This looks good to me. Thanks!

**Comment by tlambert03** [comment]
I added a note, since we don't want people looking through tests and thinking calling `worker.work()` is a good practice.

File: napari/_qt/_tests/test_threading.py (hunk 1 above)
Comment (jni) [reply napari/_qt/_tests/test_threading.py]: Lovely!

**Pull request merged**